.github/workflows/snapshot-record.yaml

```diff
@@ -0,0 +1,35 @@
+name: Snapshot-Record
+
+on: 
+  push:
+    tags:
+    - 'snapshot-record*'
+
+jobs:
+  record_snapshots:
+    name: Record snapshots
+    runs-on: macos-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
+          java-version: 11
+      - uses: ./.github/actions/gradle-cache
+        with:
+          key-prefix: gradle-test
+      - name: Snapshot tests
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 28
+          disable-animations: true
+          profile: Nexus 4
+          script: ./gradlew stream-chat-android-ui-uitests:executeScreenshotTests -Precord
+      - name: Upload screnshots
+        uses: actions/upload-artifact@v2
+        with:
+          name: screenshots
+          path: ${{ github.workspace }}/stream-chat-android-ui-uitests/screenshots/
```

Comment:
Where are you uploading those screenshots?
They need to be pushed to the repository, right?
Instead of this step, you should create a new commit over the same branch than the PR is started from.
You can use [this action](https://github.com/marketplace/actions/add-commit) for that

Reply:
The idea is not to create a PR with snapshots but to create an action that the developer can use to record new pictures and add then to their PR with new snapshot tests